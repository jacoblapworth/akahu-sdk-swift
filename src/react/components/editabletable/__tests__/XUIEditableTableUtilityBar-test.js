import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { axe, toHaveNoViolations } from 'jest-axe';

import XUIEditableTableUtilityBar from '../XUIEditableTableUtilityBar';

Enzyme.configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

describe('<XUIEditableTableUtilityBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<XUIEditableTableUtilityBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('composes the className correctly', () => {
    const wrapper = shallow(<XUIEditableTableUtilityBar className="test-classname" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass accessibility testing', async () => {
    const wrapper = mount(
      <table>
        <tfoot>
          <XUIEditableTableUtilityBar />
        </tfoot>
      </table>,
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
});
