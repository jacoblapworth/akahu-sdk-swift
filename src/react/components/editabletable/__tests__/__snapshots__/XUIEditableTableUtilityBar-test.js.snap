// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XUIEditableTableUtilityBar /> composes the className correctly 1`] = `
<tr
  className="xui-editabletableutilitybar test-classname"
>
  <td
    className="xui-editabletableutilitybar--cell"
    colSpan={0}
  >
    <div
      className="xui-editabletableutilitybar--cell--wrapper"
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </td>
</tr>
`;

exports[`<XUIEditableTableUtilityBar /> renders correctly 1`] = `
<tr
  className="xui-editabletableutilitybar"
>
  <td
    className="xui-editabletableutilitybar--cell"
    colSpan={0}
  >
    <div
      className="xui-editabletableutilitybar--cell--wrapper"
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </td>
</tr>
`;
